File: app/src/main/java/com/skybox/seven/covid/ui/prevention/PreventionFragment.java
package com.skybox.seven.covid.ui.prevention;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;
import androidx.navigation.Navigation;

import com.airbnb.epoxy.EpoxyRecyclerView;
import com.skybox.seven.covid.R;
import com.skybox.seven.covid.data.entities.Advice;
import com.skybox.seven.covid.epoxy.prevention.PreventionController;
import com.skybox.seven.covid.ui.common.ShareDialogDialog;
import com.skybox.seven.covid.util.SpaceItemDecorator;

import dagger.hilt.android.AndroidEntryPoint;


@AndroidEntryPoint
public class PreventionFragment extends Fragment implements PreventionController.PreventionCallback, View.OnClickListener {

    private PreventionController controller;
    private PreventionViewModel viewModel;

    public PreventionFragment() { }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        viewModel = new ViewModelProvider(requireActivity()).get(PreventionViewModel.class);
        controller = new PreventionController(this);

        viewModel.adviceList.observe(this, preventionModel -> controller.setData(false,preventionModel));

        viewModel.getAdviceList();
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {

        View v = inflater.inflate(R.layout.fragment_prevention, container, false);
        EpoxyRecyclerView recyclerView = v.findViewById(R.id.PreventionRecyclerView);

        recyclerView.addItemDecoration(new SpaceItemDecorator(20, true, false));
        recyclerView.setController(controller);

        Button backBtn = v.findViewById(R.id.backBtnn);
        backBtn.setOnClickListener((View.OnClickListener) this);

        return v;
    }

    @Override
    public void preventionClick(Advice advice) {
        ShareDialogDialog.newInstance(advice).show(getChildFragmentManager(), advice.getTitle());
    }

    public void onClick(View view){
        Navigation.findNavController(requireActivity(), R.id.nav_host_fragment).navigateUp();
    }
}

File: app/src/main/java/com/skybox/seven/covid/ui/prevention/PreventionFragment.kt
package com.skybox.seven.covid.ui.prevention

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.fragment.app.activityViewModels
import androidx.fragment.app.viewModels
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import androidx.navigation.Navigation
import com.skybox.seven.covid.R
import com.skybox.seven.covid.data.entities.Advice
import com.skybox.seven.covid.databinding.FragmentPreventionBinding
import com.skybox.seven.covid.epoxy.prevention.PreventionController
import com.skybox.seven.covid.epoxy.prevention.PreventionController.PreventionCallback
import com.skybox.seven.covid.ui.share.ShareDialogViewModel
import com.skybox.seven.covid.util.SpaceItemDecorator
import dagger.hilt.android.AndroidEntryPoint

@AndroidEntryPoint
class PreventionFragment : Fragment(), PreventionCallback, View.OnClickListener {
    private var controller: PreventionController = PreventionController(this)
    private val viewModel: PreventionViewModel by viewModels()
    private val shareViewModel: ShareDialogViewModel by activityViewModels()

    private lateinit var binding: FragmentPreventionBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        viewModel.adviceList.observe(this, Observer { preventionModel: List<Advice?> -> controller.setData(false, preventionModel) })
        viewModel.getAdviceList()
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        binding = FragmentPreventionBinding.inflate(inflater, container, false)
        binding.PreventionRecyclerView.addItemDecoration(SpaceItemDecorator(20, true, false))
        binding.PreventionRecyclerView.setController(controller)
        binding.backBtnn.setOnClickListener(this)
        return binding.root
    }

    override fun preventionClick(advice: Advice) {
        shareViewModel.advice.value = advice
        shareViewModel.open.value = true
    }

    override fun onClick(view: View) {
        Navigation.findNavController(requireActivity(), R.id.nav_host_fragment).navigateUp()
    }

    override fun onDestroy() {
        shareViewModel.open.value = false
        super.onDestroy()
    }
}